modules/bbmap/clumpify/main.nf

```diff
@@ -0,0 +1,37 @@
+process BBMAP_CLUMPIFY {
+    tag "$meta.id"
+    label 'process_high_memory'
+
+    conda (params.enable_conda ? "bioconda::bbmap=38.98" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bbmap:38.98--h5c4e2a8_1' :
+        'quay.io/biocontainers/bbmap:38.98--h5c4e2a8_1' }"
+
+    input:
+    tuple val(meta), path(reads)
+
+    output:
+    tuple val(meta), path('*.fastq.gz'), emit: reads
+    tuple val(meta), path('*.log')     , emit: log
+    path "versions.yml"                , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def raw      = meta.single_end ? "in=${reads[0]}" : "in1=${reads[0]} in2=${reads[1]}"
+    def clumped  = meta.single_end ? "out=${prefix}.clumped.fastq.gz" : "out1=${prefix}_1.clumped.fastq.gz out2=${prefix}_2.clumped.fastq.gz"
+    """
+    clumpify.sh \\
+        $raw \\
+        $clumped \\
+        $args \\
+        &> ${prefix}.clumpify.log
```

Comment:
Is this a single threaded program? No `task.cpus` usable?

Reply:
This is a java based program - I just followed how other bbmap modules were written - https://github.com/nf-core/modules/blob/master/modules/bbmap/bbduk/main.nf

Reply:
I checked over the options. This one appears to be single threaded.